test/config/dump.c

```diff
@@ -0,0 +1,368 @@
+/**
+ * @file
+ * Test code for the Config Dump functions
+ *
+ * @authors
+ * Copyright (C) 2017-2018 Richard Russon <rich@flatcap.org>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define TEST_NO_MAIN
+#include "acutest.h"
+#include "config.h"
+#include "mutt/mutt.h"
+#include "config/common.h"
+#include "config/lib.h"
+
+static bool VarApple;
+static bool VarBanana;
+static short VarCherry;
+static struct Address *VarElderberry;
+static char *VarFig;
+static long VarGuava;
+static short VarHawthorn;
+static struct MbTable *VarIlama;
+static char *VarJackfruit;
+static char VarKumquat;
+static struct Regex *VarLemon;
+static short VarMango;
+static char *VarNectarine;
+
+// clang-format off
+static struct ConfigDef Vars[] = {
+  { "Apple",      DT_BOOL,                 0, &VarApple,      false,                       NULL },
+  { "Banana",     DT_BOOL,                 0, &VarBanana,     true,                        NULL },
+  { "Cherry",     DT_NUMBER,               0, &VarCherry,     0,                           NULL },
+  { "Damson",     DT_SYNONYM,              0, NULL,           IP "Cherry",                 NULL },
+  { "Elderberry", DT_ADDRESS,              0, &VarElderberry, IP "elderberry@example.com", NULL },
+  { "Fig",        DT_COMMAND|DT_NOT_EMPTY, 0, &VarFig,        IP "fig",                    NULL },
+  { "Guava",      DT_LONG,                 0, &VarGuava,      0,                           NULL },
+  { "Hawthorn",   DT_MAGIC,                0, &VarHawthorn,   1,                           NULL },
+  { "Ilama",      DT_MBTABLE,              0, &VarIlama,      0,                           NULL },
+  { "Jackfruit",  DT_PATH,                 0, &VarJackfruit,  IP "/etc/passwd",            NULL },
+  { "Kumquat",    DT_QUAD,                 0, &VarKumquat,    0,                           NULL },
+  { "Lemon",      DT_REGEX,                0, &VarLemon,      0,                           NULL },
+  { "Mango",      DT_SORT,                 0, &VarMango,      1,                           NULL },
+  { "Nectarine",  DT_STRING,     F_SENSITIVE, &VarNectarine,  IP "nectarine",              NULL },
+  { NULL },
+};
+// clang-format on
+
+void mutt_pretty_mailbox(char *buf, size_t buflen)
+{
+}
```

Comment:
Let's discard this since it's empty.

Reply:
That's used by `dump_config()`
The fact that it doesn't do anything doesn't matter for the purposes of testing.